d2graph/d2graph.go

```diff
@@ -1059,6 +1098,20 @@ func (g *Graph) SetDimensions(mtexts []*d2target.MText, ruler *textmeasure.Ruler
 
 		case d2target.ShapeText, d2target.ShapeCode:
 		}
+
+		switch shapeType {
+		case d2target.ShapeClass,
+			d2target.ShapeSQLTable,
+			d2target.ShapeCode,
+			d2target.ShapeText:
+			if float64(setWidth) > obj.Width {
+				obj.Width = float64(setWidth)
+			}
+			if float64(setHeight) > obj.Height {
+				obj.Height = float64(setHeight)
+			}
```

Comment:
u need to do this again?

Reply:
this is for class, sql_table, etc which don't have their dimensions set until this point

Reply:
refactored to make this whole function clearer